src/Search.f90

```diff
@@ -0,0 +1,286 @@
+!++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++!
+!                          Futility Development Group                          !
+!                             All rights reserved.                             !
+!                                                                              !
+! Futility is a jointly-maintained, open-source project between the University !
+! of Michigan and Oak Ridge National Laboratory.  The copyright and license    !
+! can be found in LICENSE.txt in the head directory of this repository.        !
+!++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++!
+!> @brief A Fortran 2003 module implementing some basic search algorithms.
+!>
+!> @par Module Dependencies
+!>  - @ref IntrType "IntrType": @copybrief IntrType
+!>
+!> @author Mitchell Young
+!>    @date 3/30/2017
+!>
+!++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++!
+MODULE Search
+  USE IntrType
+  USE Strings
+  IMPLICIT NONE
+
+  PRIVATE
+
+  PUBLIC :: upperBound
+  PUBLIC :: lowerBound
+  PUBLIC :: find
+  PUBLIC :: binarySearch
+
+  !> @brief Return the index of the first element of a list that compares
```

Comment:
Tried building with Doxygen, but there is no Doxygen file for building the Futility docs standalone. There should be, so I created Issue #15. As for the location, I strongly prefer that the documentation live in one place. As a generic interface, the provided documentation should apply to all implementations, and as a user of the library, one shouldn't have to dig down to a specific implementation to see what a generic interface does.